Ktb 19 refactoring serialization

DenisGubernatorov opened the pull request.

DenisGubernatorov committed:
KTB-19-refactoring-serialization
1) Доработка регулярки для получения  chatId с отрицательными значениями

DenisGubernatorov committed:
KTB-19-refactoring-serialization
1) Подключение плагина сериализации
2) Внедрение зависимости для json-сериализации
3) Перевод работы бота на использование объектов json
4) Удаление более неиспользуемых методов

DenisGubernatorov committed:
KTB-19-refactoring-serialization
1) Перевод отправки методов отправки сообщений на использование json

dzenonce commented on src/main/kotlin/Telegram.kt:
```diff
@@ -1,79 +1,114 @@
     val botService = TelegramBotService(args[0])
-    var updateId = 0
+    var lastUpdateId = 0L
+    val json =
```
Давай `json` сделаем полем внутри `TelegramBotService()` и не будем передавать через методы. Ведь парсинг `json`-а это по сути задача сервиса, нам не важно в каком виде приходят данные от Telegram API.

dzenonce reviewed (changes requested):
Не понятно, как выйти из режима обучения, если только не через `/start`. Давай добавим кнопку "В меню".

DenisGubernatorov committed:
KTB-19-refactoring-serialization
1) Перенос создания и использования объекта json  в TelegramBotService

DenisGubernatorov committed:
KTB-19-refactoring-serialization
1) В варианты ответа добавлен вариант "Выход", по которой происходит выход из режима изучения слов и бот ждет перезапуска

DenisGubernatorov commented:
> Не понятно, как выйти из режима обучения, если только не через `/start`. Давай добавим кнопку "В меню".
Да. Дейстивтельно. Походу эту кнопку забыли реализовать для телеграмм бота. Я посмотрел уроки дальше, до дополнительных этапов. Там ничего такого нет.
